polyfill/lib/date.mjs

```diff
@@ -123,13 +129,18 @@ export class Date {
     if (!ES.IsDate(this)) throw new TypeError('invalid receiver');
     return new Intl.DateTimeFormat(...args).format(this);
   }
-  withTime(timeLike) {
+  withTime(time) {
     if (!ES.IsDate(this)) throw new TypeError('invalid receiver');
+    if (!ES.IsTime(time)) throw new TypeError('invalid Temporal.Time object');
```

Comment:
There was already a test for this in `date.mjs`, but it was using `{ hours: 11, minutes: 30, seconds: 23 }`. Let's add a test that would have failed before as well. (Same for `withDate`.)

Reply:
The existing test was actually a bug, since it called withTime on the DateTime object that resulted from a previous call to withTime, and therefore always threw an exception because the method didn't exist. I've fixed that up and added a similar test to Time.withDate.